File: lv1/달리기 경주.py
def solution(players, callings):
    rank = {player: index for index, player in enumerate(players)}

    for i in callings:
        idx = rank[i] # 현재 등수
        pre_idx = idx - 1 #앞에 있는 선수 등수
        players[idx], players[pre_idx] = players[pre_idx], players[idx] # 서로 바꾸기
        rank[players[idx]] = idx
        rank[players[pre_idx]] = pre_idx

    return players

File: lv1/달리기 경주.java
import java.util.*;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        Map<String, Integer> rank = new HashMap<String, Integer>();		  
		  for (int i = 0; i < players.length; i++) rank.put(players[i], i);
		  
		  for (String player : callings) {
			  int playerRanking = rank.get(player);
			  
			  String frontplayer = players[playerRanking-1];
			  rank.replace(frontplayer, playerRanking);			  
			  rank.replace(player, playerRanking-1);
			  
			  players[playerRanking] = frontplayer;
			  players[playerRanking-1] = player;
		  }
        
        return players;
    }
}
